### PR TITLE
[PREVIEW] initial setup

### PR DESCRIPTION
**Pull request by r-kruk-kainos**

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+@Library("Infrastructure")
+
+def product = "snl"
+def component = "notes"
+
+withPipeline("java", product, component) {
+  enableDbMigration()
+  after('checkout') {
+    echo 'sln-notes checked out'
```
Comment (leszekgu) [inline Jenkinsfile_CNP]: sln -> snl

File: build.gradle [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    csv.enabled = false
+    xml.destination file("${project.buildDir}/reports/jacoco/test/jacocoTestReport.xml")
+  }
+}
+
+project.tasks['sonarqube'].dependsOn test
+sonarqube {
+  properties {
+    property "sonar.projectName", "Reform :: Scheduling & Listing :: snl-notes POC build extension"
+    property "sonar.projectKey", "snl-events"
```
Comment (leszekgu) [inline build.gradle]: snl-notes as a key

File: build.gradle [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    includeCategories 'uk.gov.hmcts.reform.sandl.snlnotes.functional.SmokeTest'
+  }
+}
+
+task functional(type: Test, description: 'Runs the functional tests.', group: 'Verification') {
+  testClassesDirs = sourceSets.functional.output.classesDirs
+  classpath = sourceSets.functional.runtimeClasspath
+}
+
+task integrationFakeRulesTests(type: Test, description: 'Runs the integration tests.', group: 'Verification') {
```
Comment (leszekgu) [inline build.gradle]: we don't need fakeRules related things

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+task integrationTests(type: Test, description: 'Runs the integration tests.', group: 'Verification') {
+  testClassesDirs = sourceSets.integrationSources.output.classesDirs
+  classpath = sourceSets.integrationSources.runtimeClasspath
+
+  useJUnit {
+    includeCategories 'uk.gov.hmcts.reform.sandl.snlnotes.testcategories.IntegrationTest'
+  }
+}
+
+test.finalizedBy integrationFakeRulesTests
```
Comment (leszekgu) [inline build.gradle]: change to just 'integrationTests'

File: docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,55 @@
+version: '2.1'
+
+services:
+  snl-notes-db:
+    container_name: snl-notes-db
+    build: ./docker/snl-notes-db
+    networks:
+      - snl-notes-network
+    ports:
+      - "5433:5432"
```
Comment (leszekgu) [inline docker-compose.yml:10]: ports might be an issue in our case when snl-events-db will be running, please check if both docker can be started at the same time on local machine.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+    environment:
+      - POSTGRES_USER=snluser
+      - POSTGRES_PASSWORD=snlpass
+      - POSTGRES_DB=snl
+  snl-wait-for-db:
+    container_name: snl-wait-for-db
```
Comment (leszekgu) [inline docker-compose.yml]: check with snl-events running - will it take snl-events information or create a separate container

File: infrastructure/variables.tf [hunk 7]
```diff
@@ -0,0 +1,57 @@
+variable "product" {
+  type        = "string"
+  default     = "snl"
+  description = "Scheduling and Listing "
+}
+variable "component" {
+  type        = "string"
+  default     = "events"
```
Comment (leszekgu) [inline infrastructure/variables.tf]: notes

**Review by leszekgu — CHANGES_REQUESTED** [review]
as in comments

File: docker-compose.yml (hunk 5 above)
Comment (r-kruk-kainos) [reply docker-compose.yml:10]: It works :)

**Review by leszekgu — APPROVED** [review]

**Pull request merged**